api/src/test/java/com/epam/pipeline/manager/notification/NotificationAspectTest.java

```diff
@@ -16,145 +16,109 @@
-        Assert.assertEquals(TaskStatus.SUCCESS.name(), message.getTemplateParameters().get("status"));
-
-        Mockito.verify(pipelineRunDao).updateRunStatus(Mockito.any());
-        Mockito.verify(runStatusManager).saveStatus(Mockito.any());
+        final ArgumentCaptor<NotificationMessage> captor =
+            ArgumentCaptor.forClass(NotificationMessage.class);
+        verify(mockMonitoringNotificationDao).createMonitoringNotification(captor.capture());
+
+        final NotificationMessage capturedMessage = captor.getValue();
+        Assert.assertEquals(pipelineUser.getId(), capturedMessage.getToUserId());
+        Assert.assertEquals(TaskStatus.SUCCESS.name(),
+            capturedMessage.getTemplateParameters().get("status"));
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Throwable.class)
     public void testNotifyRunStatusChangedNotActiveIfStatusNotConfiguredForNotification() {
-        PipelineRun run = new PipelineRun();
-        run.setStatus(TaskStatus.PAUSED);
-        run.setOwner(testOwner.getUserName());
-        run.setStartDate(new Date());
+        final NotificationSettings settings = NotificationCreatorUtils.getNotificationSettings(ID);
+        final PipelineRun run = PipelineCreatorUtils.getPipelineRun(TaskStatus.PAUSED);
+        doReturn(settings).when(mockNotificationSettingsDao).loadNotificationSettings(any());
+        mockRole();
 
         pipelineRunManager.updatePipelineStatus(run);
-        List<NotificationMessage> messages = monitoringNotificationDao.loadAllNotifications();
-        Assert.assertTrue(messages.isEmpty());
+
+        verify(mockMonitoringNotificationDao, never()).createMonitoringNotification(any());
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Throwable.class)
     public void testNotifyRunStatusChangedActiveIfSettingsDoesntHaveStatusesConfigured() {
-        NotificationSettings toUpdate = notificationSettingsDao
-                .loadNotificationSettings(NotificationSettings.NotificationType.PIPELINE_RUN_STATUS.getId());
-        toUpdate.setStatusesToInform(Collections.emptyList());
-        notificationSettingsDao.updateNotificationSettings(toUpdate);
-        PipelineRun run = new PipelineRun();
-        run.setStatus(TaskStatus.PAUSED);
-        run.setOwner(testOwner.getUserName());
-        run.setStartDate(new Date());
+        final NotificationSettings settings = NotificationCreatorUtils.getNotificationSettings(ID);
+        final PipelineRun run = PipelineCreatorUtils.getPipelineRun(TaskStatus.PAUSED);
+        doReturn(settings).when(mockNotificationSettingsDao).loadNotificationSettings(any());
+        doReturn(pipelineUser).when(mockUserDao).loadUserByName(any());
+        settings.setStatusesToInform(Collections.emptyList());
+        doReturn(settings).when(mockNotificationSettingsDao).updateNotificationSettings(any());
```

Comment:
It seems that this mock is not obsolete now. (We were needed to call `updateNotificationSettings` method because we were creating `settings` object into the `setup` method. We don't need such logic now)